src/citrine/resources/material_run.py

```diff
@@ -166,3 +167,54 @@ def get_history(self, scope, id) -> Type[MaterialRun]:
         model = loads(json.dumps([data['context'], data['root']], cls=TaurusEncoder))
         # Convert taurus objects into citrine-python objects
         return MaterialRun.build(model)
+
+    def with_spec(self,
```

Comment:
There is precedent in citrine-python (e.g. https://github.com/CitrineInformatics/citrine-python/blob/391283f46bc8cead857df4a25c04b8b319a84e7f/src/citrine/resources/data_concepts.py#L540) to use the naming convention `filter_by_*` for methods like this. Thoughts on calling this `filter_by_spec`?

Reply:
That's a solid point. I forgot about these. I am on the side of keeping this consistent by using filter_by_spec.